--- a/drivers/net/dsa/lantiq_gswip.c
+++ b/drivers/net/dsa/lantiq_gswip.c
@@ -1874,6 +1874,16 @@ static int gswip_probe(struct platform_d
 			dev_err(dev, "gphy fw probe failed\n");
 			return err;
 		}
+
+		/* The standalone PHY11G requires 300ms to be fully
+		 * initialized and ready for any MDIO communication after being
+		 * taken out of reset. For the SoC-internal GPHY variant there
+		 * is no (known) documentation for the minimum time after a
+		 * reset. Use the same value as for the standalone variant as
+		 * some users have reported internal PHYs not being detected
+		 * without any delay.
+		 */
+		msleep(300);
 	}
 
 	/* bring up the mdio bus */
